doc/developer/design/20220516_transactional_consistency.md

```diff
@@ -0,0 +1,446 @@
+chapters 7 and 9, of [Designing Data-Intensive Applications](https://dataintensive.net/). This document focuses on
+transactional consistency levels included in the ANSI SQL spec.
+
+#### Read Uncommitted
+
+See [https://jepsen.io/consistency/models/read-uncommitted](https://jepsen.io/consistency/models/read-uncommitted) for
+an in depth formal discussion of Read Uncommitted. Read Uncommitted prevents Dirty Writes, which means that one
+transaction cannot update the values of another transactions uncommitted values.
+
+This is solved by disallowing updates in transactions and by write-only transactions buffering their writes until
+commit. No update can view the buffered data until after the write-only transaction has committed. While the write-only
+transaction is committing, no other queries can execute. Once the commit is complete, all writes are guaranteed to be
+durable.
+
+#### Read Committed
+
+See [https://jepsen.io/consistency/models/read-committed](https://jepsen.io/consistency/models/read-committed) for an in
+depth formal discussion of Read Committed. Read Committed prevents Dirty Reads, which means that one transaction cannot
+view the uncommitted data of another transaction.
+
+This property holds from how Materialize satisfies [Read Uncommitted](#Read Uncommitted). No read can view uncommitted
+buffered data until after the commit is complete.
+
+#### Repeatable Read
+
+See [https://jepsen.io/consistency/models/repeatable-read](https://jepsen.io/consistency/models/repeatable-read) for an
+in depth formal discussion of Repeatable Reads. Repeatable Reads prevents Fuzzy/Non-Repeatable Reads, which means that
+once a transactions reads some value for an object, it will always read that same value, regardless if it’s been
+overwritten by another transaction.
+
+Differential Dataflow provides us with some useful properties to achieve this.
+
+1. All reads and writes take place at a provided timestamp.
+2. All reads at timestamp `ts` will return all writes at timestamp ≤ `ts`.
+3. All reads at timestamp `ts` will not return any of the writes at timestamps > `ts`.
+
+As described in the [Timestamp Selection](#Timestamp Selection) section, whenever a read happens at timestamp `ts`, all
+subsequent writes will happen at a timestamp > `ts`. Since a read transaction performs all of it’s read at the same
+timestamp, we can guarantee that subsequent reads will always read the same value since all writes will happen at a
+later timestamp.
+
+#### Serializable
+
+See [https://jepsen.io/consistency/models/serializable](https://jepsen.io/consistency/models/serializable) for an in
+depth formal discussion of Serializable. Serializability prevents Phantom Reads, which means that once a transaction
+reads some value that satisfy some predicate, it will always read the same values, regardless if some other transaction
+has made a write that affects the predicate. Informally this means that you can apply some total order to all
+transactions.
```

Comment:
I took this language from the Jepsen consistency documents (https://jepsen.io/consistency/models/serializable). 
> Informally, serializability means that transactions appear to have occurred in some total order.

My understanding is that the SQL Spec never actually mentions total ordering and describes these isolation levels as the absence of certain anomalies.

Should I change this to say "Additionally" instead of "Informally".?